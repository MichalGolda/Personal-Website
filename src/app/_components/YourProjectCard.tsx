import { PrimaryButton } from "./Button";

export default function YourProjectCard() {
  return (
    <div className="w-full h-full bg-secondary flex items-center justify-center">
      <div className="flex flex-col gap-y-4 items-center">
        <h4 className="text-white font-bold text-2xl">
          Miejsce na twój projekt
        </h4>
        <PrimaryButton className="w-fit">Skontaktuj się</PrimaryButton>
      </div>
    </div>
  );
}
