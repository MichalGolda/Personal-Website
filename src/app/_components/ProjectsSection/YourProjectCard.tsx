import { PrimaryButton } from "@/app/_components";

export default function YourProjectCard() {
  return (
    <div className="w-full h-full bg-secondary flex items-center justify-center">
      <div className="flex flex-col gap-y-4 items-center">
        <h4 className="text-white font-bold text-2xl">
          Miejsce na twój projekt
        </h4>
        <a href="#contact">
          <PrimaryButton className="w-fit">Skontaktuj się</PrimaryButton>
        </a>
      </div>
    </div>
  );
}
