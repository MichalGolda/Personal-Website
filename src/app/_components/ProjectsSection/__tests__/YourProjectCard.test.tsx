import "@testing-library/jest-dom";
import { render, screen } from "@testing-library/react";
import YourProjectCard from "../YourProjectCard";

describe("YourProjectCard component", () => {
  it("renders without errors", () => {
    render(<YourProjectCard />);

    const title = screen.getByRole("heading");
    const link = screen.getByRole("link");
    const button = screen.getByText("Skontaktuj się");

    expect(title).toBeInTheDocument();
    expect(title.tagName).toBe("H4");
    expect(title.textContent).toBe("Miejsce na twój projekt");

    expect(link).toBeInTheDocument();
    expect(link).toHaveAttribute("href", "#contact");

    expect(button).toBeInTheDocument();
  });

  it("renders unchanged", () => {
    const { container } = render(<YourProjectCard />);

    expect(container).toMatchSnapshot();
  });
});
