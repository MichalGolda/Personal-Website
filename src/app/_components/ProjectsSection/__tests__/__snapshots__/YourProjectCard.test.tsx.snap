// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`YourProjectCard component renders unchanged 1`] = `
<div>
  <div
    class="w-full h-full bg-secondary flex items-center justify-center"
    data-aos="zoom-in"
    data-aos-duration="500"
  >
    <div
      class="flex flex-col gap-y-4 items-center"
    >
      <h4
        class="text-white text-lg md:text-xl lg:text-2xl font-bold"
      >
        Miejsce na twój projekt
      </h4>
      <a
        href="#contact"
      >
        <button
          class="py-4 px-8 font-bold text-sm md:text-base transition-all duration-500 disabled:bg-lightGrey disabled:text-body bg-primary text-white hover:bg-primary/90 active:bg-primary/80 border border-transparent border-solid w-fit"
        >
          Skontaktuj się
        </button>
      </a>
    </div>
  </div>
</div>
`;
